src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MeanQualityByCycleSpark.java

```diff
@@ -0,0 +1,273 @@
+import org.broadinstitute.hellbender.cmdline.argumentcollections.OptionalIntervalArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.ReadInputArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.RequiredReadInputArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.picard.analysis.MeanQualityByCycle;
+import org.broadinstitute.hellbender.utils.R.RScriptExecutor;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Program to generate a data table and chart of mean quality by cycle from a
+ * BAM file.  Works best on a single lane/run of data, but can be applied to
+ * merged BAMs - the output may just be a little confusing.
+ *
+ * This is the Spark implementation.
+ *
+ */
+@CommandLineProgramProperties(
+summary = "Program to generate a data table and pdf chart of " +
+        "mean base quality by cycle from a SAM or BAM file.  Works best on a single lane/run of data, but can be applied to " +
+        "merged BAMs. Uses R to generate chart output.",
+        oneLineSummary = "Writes mean quality by cycle for a SAM or BAM file",
+        programGroup = SparkProgramGroup.class
+)
+public final class MeanQualityByCycleSpark extends SparkCommandLineProgram {
+
+    private static final long serialVersionUID = 1l;
+    @ArgumentCollection
+    public ReadInputArgumentCollection readArguments= new RequiredReadInputArgumentCollection();;
+
+    @ArgumentCollection
+    public IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = true)
+    public String out;
+
+    @Argument(shortName="CHART", doc="A file (with .pdf extension) to write the chart to.", optional=true)
+    public File CHART_OUTPUT;
+
+    @Argument(doc="If set to true, calculate mean quality over aligned reads only.")
+    public boolean ALIGNED_READS_ONLY = false;
+
+    @Argument(doc="If set to true calculate mean quality over PF reads only.")
+    public boolean PF_READS_ONLY = false;
+
+    @Argument(doc = "Should an output plot be created")
+    public boolean PRODUCE_PLOT = false;
```

Comment:
I think we can use the `CHART_OUTPUT != null` instead of having a separate `PRODUCE_PLOT`.